includes/class-edd-logging.php

```diff
@@ -160,6 +166,7 @@ public function insert_log( $log_data = array(), $log_meta = array() ) {
 			'object_type' => isset( $args['log_type'] )
 				? $args['log_type']
 				: null,
+			'user_id'     => ! empty( $args['post_author'] ) ? intval( $args['post_author'] ) : 0
```

Comment:
So I'm thinking out loud here and looking at the `insert_log` method in 2.x and the `post_author` is never actually explicitly set in there. Since it uses `wp_insert_post`, the `post_author` was set to the current user ID as the default. So maybe this needs to use the current user ID as the default, skip the `post_author` as I don't think anyone would have been explicitly setting it, but then allow anyone still using this method to explicitly set the `user_id` in a new parameter (because I believe they would have had to be setting this as a new/separate parameter anyway?). I feel like I'm rambling--basically what if we set the default to the current user ID, because I think that would be consistent, but allow a [new] parameter to be passed to any log, not just the api requests. I guess the download log would ignore it if we went that route.

Reply:
How about this: https://github.com/easydigitaldownloads/easy-digital-downloads/pull/8384/commits/76f11b86fe3d840727fa46beca5875cb03f5d404

I used `$log_meta['user']` as that was already used by API request logs, so it felt consistent.